packages/storybook-utils/src/events.ts

```diff
@@ -0,0 +1,777 @@
+import type { Events } from "vue";
+
+// prettier-ignore
+type EventName<T extends Event> = 
+  T extends ClipboardEvent ? "ClipboardEvent"
+: T extends WheelEvent ? "WheelEvent"
+: T extends PointerEvent ? "PointerEvent"
+: T extends DragEvent ? "DragEvent"
+: T extends MouseEvent ? "MouseEvent"
+: T extends CompositionEvent ? "CompositionEvent"
+: T extends FocusEvent ? "FocusEvent"
+: T extends KeyboardEvent ? "KeyboardEvent"
+: T extends TouchEvent ? "TouchEvent"
+: T extends AnimationEvent ? "AnimationEvent"
+: T extends TransitionEvent ? "TransitionEvent"
+: "Event";
+
+/**
+ * Removes a prefix from a string type.
+ * @example ```ts
+ * type Trimmed = TrimStart<"onclick", "on">;
+ * const trimmed: Trimmed = "click";
+ * ```
+ */
+type TrimStart<
+  TString extends string,
+  Prefix extends string,
+> = TString extends `${Prefix}${infer Base}` ? Base : TString;
```

Comment:
```suggestion
  TPrefix extends string,
> = TString extends `${TPrefix}${infer TBase}` ? TBase : TString;
```

Isn't it a convention to prefix generic types with T?